docs/core/v3.0.rst

```diff
@@ -0,0 +1,1762 @@
+    integer data type defines binary representations for all integers
+    in the range −2,147,483,648 to 2,147,483,647. This specification
+    only defines a limited set of data types, but extensions
+    may define other data types.
+
+.. _chunk:
+.. _chunks:
+
+*Chunk*
+
+    An array_ is divided into a set of chunks, where each chunk is a
+    hyperrectangle defined by a tuple of intervals, one for each
+    dimension_ of the array_. The chunk shape is the tuple of interval
+    lengths, and the chunk size (i.e., number of elements_ contained
+    within the chunk) is the product of its interval lengths.
+
+    The chunk shape elements are non-zero when the corresponding dimensions of
+    the arrays are of non-zero length.
+
+.. _grid:
+.. _grids:
+
+*Grid*
+
+    The chunks_ of an array_ are organised into a grid. This
+    specification only considers the case where all chunks_ have the
+    same chunk shape and the chunks form a regular grid. However,
+    extensions may define other grid types such as
+    rectilinear grids.
+
+.. _memory layout:
+.. _memory layouts:
+
+*Memory layout*
+
+    An array_ is associated with a memory layout which defines how to
+    construct a binary representation of a single chunk_ by organising
+    the binary values of the elements_ within the chunk_ into a single
+    contiguous sequence of bytes. This specification defines two types
+    of memory layout based on "C" (row-major) and "F" (column-major)
+    ordering of elements_, but extensions may define other
+    memory layouts.
+
+.. _compressor:
+.. _compressors:
+
+*Compressor*
+
+    An array_ may be associated with a compressor, which is a codec_
+    that transforms the binary representation of a chunk_ in some way,
+    usually to reduce data size.
+
+.. _codec:
+.. _codecs:
+
+*Codec*
+
+    A codec is a pair of algorithms which transform binary data in
+    some way and are used to encode and decode chunks_. This
+    specification defines a codec interface which comprises
+    a pair of operations, one to perform the transformation (encode),
+    the other to reverse the transformation (decode). This
+    specification only considers the case where a codec is used as a
+    compressor_, but extensions may extend the chunk encoding
+    process, for example, to add support for one or more filter codecs
+    to be applied prior to compression.
+
+.. _metadata document:
+.. _metadata documents:
+
+*Metadata document*
+
+    Each array_ in a hierarchy_ is represented by a metadata document,
+    which is a machine-readable document containing essential
+    processing information about the node. For example, an array_
+    metadata document will specify the number of dimensions_, shape_,
+    `data type`_, grid_, `memory layout`_ and compressor_ for that
+    array_.
+
+    Groups can have a optional metadata document which provide extra
+    information about a group.
+
+.. _store:
+.. _stores:
+
+*Store*
```

Comment:
It could be helpful for understanding to give an example of what a store is. All the other definitions are quite clear (at least to me); this one not so much. I assume it means (an abstraction of) the filesystem implementation e.g. local, s3, abs etc.

Reply:
I tried to clarify that in https://github.com/alimanfoo/zarr-specs/pull/1 using some examples from the v2 spec. @constantinpape Please let me know if this is not clear enough or you'd like to see further changes (also directly in https://github.com/alimanfoo/zarr-specs/pull/1 if you want).

Reply:
Those changes have been merged in #163, marking as resolved for now.